packages/fxa-settings/src/models/reliers/rellier.ts

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import "reflect-metadata";
+import { ContextValidation as V, bind, validateContext, ContextKeyTransforms as T, ContextProvider } from '../../lib/context';
+import { Context } from "../../lib/context/interfaces/context";
+
+/**
+ * A convenience interface for easy mocking / testing. Use this type in components.
+ */
+export interface RelierData
+{
+  context: string;
+  name: string;
+  email: string | undefined;
+  entrypoint: string | undefined;
+  entrypointExperiment: string | undefined;
+  entrypointVariation: string | undefined;
+  resetPasswordConfirm: boolean | undefined;
+  setting: string | undefined;
+  service: string | undefined;
+  serviceName: string | undefined;
+  subscriptionProductId: string | undefined;
+  subscriptionProductName: string | undefined;
+  style: string | undefined;
+  uid: string | undefined;
+  utmCampaign: string | undefined;
+  utmContent: string | undefined;
+  utmMedium: string | undefined;
+  utmSource: string | undefined;
+  utmTerm: string | undefined;
+}
+
+/**
+ * Create a relier class that can be bound to a given context.
+ */
+export class Relier implements ContextProvider, RelierData {
+
+  @bind([V.isString, V.isRequired])
+  context: string = '';
+
+  @bind([V.isString])
+  name: string = '';
+
+  @bind([V.isString])
+  email: string | undefined;
+
+  @bind([V.isString])
+  entrypoint: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  entrypointExperiment: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  entrypointVariation: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  resetPasswordConfirm: boolean | undefined;
+
+  @bind([V.isString], "setting")
+  setting: string | undefined;
+
+  @bind([V.isString], "service")
+  service: string | undefined;
+
+  // TODO: Double check
+  @bind([V.isString], "service")
+  serviceName: string | undefined;
+
+  @bind([V.isString])
+  subscriptionProductId: string | undefined;
+
+  @bind([V.isString])
+  subscriptionProductName: string | undefined;
+
+  @bind([V.isString])
+  style: string | undefined;
+
+  @bind([V.isString])
+  uid: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmCampaign: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmContent: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmMedium: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmSource: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmTerm: string | undefined;
+
+  constructor(
+    protected readonly curContext:Context
+  ) { }
+
+  getContext() {
+    return this.curContext;
+  }
+
+  validate() {
+    validateContext(this)
+  }
+}
```

Comment:
There might be some other convenience methods needed here as well. This will become more obvious when tests are ported.